## Importing Libraries

In [2]:
#Importing libraries
import pandas as pd
import numpy as np
import os

## Importing Data

In [ ]:
#path to main folder
path = r'C:\Users\steve\Documents\11.24 Instacart Basket Analysis'
#variable storing columns to keep from df
cols_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']
#import df
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), usecols = cols_list, index_col = False)
#import products file
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'products.csv'), index_col = False)

## Analysis

In [ ]:
#print first 5 rows of data from df
df.head()

In [ ]:
#print prods first 5 rows 
df_prods.head()

In [ ]:
#print prods last 5 rows
df_prods.tail(5)

In [23]:
#print prods names of cols
df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [25]:
#print prods number of rows and columns
df_prods.shape

(49693, 5)

In [31]:
#prods stats
df_prods.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

In [33]:
#data type of department_id
df_prods.dtypes

product_id         int64
product_name      object
aisle_id           int64
department_id      int64
prices           float64
dtype: object

01. What's the maximum value of the "aisle_id" column?
   134
02. What's the data type of the "department_id" column?
   integar (up to 64 characters)